core/sdk/src/extn/client/extn_client.rs

```diff
@@ -121,17 +121,33 @@ impl ExtnClient {
     ///
     /// Also starts the thread in the processor to accept incoming requests.
     pub fn add_request_processor(&mut self, mut processor: impl ExtnRequestProcessor) {
+        let contracts = if let Some(multiple_contracts) = processor.fulfills_mutiple() {
+            multiple_contracts
+        } else {
+            vec![processor.contract()]
+        };
+        let mut atleast_one_contract_satisfied = false;
+        let mut contracts_supported = Vec::new();
+        for contract in contracts {
+            if self.sender.check_contract_fulfillment(contract) {
+                atleast_one_contract_satisfied = true;
+                let processor_string: String = processor.contract().as_clear_string();
+                info!("adding request processor {}", processor_string);
+                add_stream_processor(
+                    processor_string.clone(),
+                    processor.sender(),
+                    self.request_processors.clone(),
+                );
+                contracts_supported.push(processor_string);
+            }
+        }
```

Comment:
Rather than use a mutable vector to collect the contracts we could filter the contracts to find the suitable ones and then add them all as stream_processers.

```suggestion
        let contracts_supported = contracts
            .iter()
            .filter(|contract| self.sender.check_contract_fulfillment(contract))
            .collect();
            
        contracts_supported.for_each(|contract| {
                let processor_string: String = processor.contract().as_clear_string();
                info!("adding request processor {}", processor_string);
                add_stream_processor(
                    processor_string,
                    processor.sender(),
                    self.request_processors.clone(),
                );
        });
```

I am not sure if this suits the code style we are working towards but I find reduced nesting a little easier to read